src/mongo_logger.ts

```diff
@@ -421,13 +453,100 @@ export function stringifyWithMaxLen(
 ): string {
   let strToTruncate = '';
 
+  let currentLength = 0;
+  const maxDocumentLengthEnsurer = function maxDocumentLengthEnsurer(key: string, value: any) {
+    if (currentLength >= maxDocumentLength) {
+      return undefined;
+    }
+    // Account for root document
+    if (key === '') {
+      // Account for starting brace
+      currentLength += 1;
+      return value;
+    }
+
+    // +4 accounts for 2 quotation marks, colon and comma after value
+    currentLength += key.length + 4;
+
+    if (value == null) return value;
+
+    switch (typeof value) {
+      case 'string':
+        // +2 accounts for quotes
+        currentLength += value.length + 2;
```

Comment:
Our goal is to undershoot, but it may be worth keeping in mind that JSON will escape things that aren't really escaped in the string's content. 

```
> '\n'.length
1
> JSON.stringify('\n').length
4
```

Reply:
🤔 should we just update the comment here? I'm not sure if we can go through and check for all the escaped characters without making this expensive.